katharsis-client/src/main/java/io/katharsis/client/internal/BaseResponseDeserializer.java

```diff
@@ -0,0 +1,198 @@
+package io.katharsis.client.internal;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.katharsis.dispatcher.controller.resource.ResourceUpsert;
+import io.katharsis.queryParams.QueryParams;
+import io.katharsis.repository.RepositoryMethodParameterProvider;
+import io.katharsis.request.dto.DataBody;
+import io.katharsis.request.dto.RequestBody;
+import io.katharsis.request.path.JsonPath;
+import io.katharsis.resource.information.ResourceInformation;
+import io.katharsis.resource.registry.RegistryEntry;
+import io.katharsis.resource.registry.ResourceRegistry;
+import io.katharsis.response.BaseResponseContext;
+import io.katharsis.response.CollectionResponseContext;
+import io.katharsis.response.JsonApiResponse;
+import io.katharsis.response.ResourceResponseContext;
+import io.katharsis.utils.parser.TypeParser;
+
+/**
+ * Deerializes top-level JSON object and provides ability to include compound documents
+ * 
+ * TODO could this be merged with RequestBodyDeserializer?
+ */
+public class BaseResponseDeserializer extends JsonDeserializer<BaseResponseContext> {
+
+	private static final String INCLUDED_FIELD_NAME = "included";
+	private static final String DATA_FIELD_NAME = "data";
+	private static final String META_FIELD_NAME = "meta";
+	private static final String LINKS_FIELD_NAME = "links";
+
+	private ResourceRegistry resourceRegistry;
+	private ObjectMapper objectMapper;
+	
+	private TypeParser typeParser = new TypeParser();
+
+	public BaseResponseDeserializer(ResourceRegistry resourceRegistry, ObjectMapper objectMapper) {
+		this.resourceRegistry = resourceRegistry;
+		this.objectMapper = objectMapper;
+	}
+
+	@Override
+	public BaseResponseContext deserialize(JsonParser jp, DeserializationContext ctxt) throws IOException, JsonProcessingException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'com.fasterxml.jackson.core.JsonProcessingException' which is a subclass of 'java.io.IOException'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)